pos-print/src/test/java/com/clouway/pos/print/adapter/http/ReportServiceTest.kt

```diff
@@ -43,96 +43,103 @@ class ReportServiceTest {
+    val reply = reportService.printOperatorReport(request.mockRequest(getOperatorReport()))
 
     assertThat(reply, isStatus(480))
     assertThat(reply, contains(ErrorResponse("Device can't connect.")))
   }
 
   @Test
   fun printReportForPeriod() {
-    val periodReportDTO = ReportService.PeriodReportDTO(LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), "::any ip::", "SHORT")
-
     context.checking(object : Expectations() {
       init {
         oneOf(factory).getPrinter("::any ip::")
         will(AbstractExpectations.returnValue(printer))
-        oneOf(printer).reportForPeriod(LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), PeriodType.SHORT)
+        oneOf(printer).reportForPeriod(LocalDateTime.parse("2017-01-01T00:00:00.000Z", DateTimeFormatter.ISO_DATE_TIME), LocalDateTime.parse("2017-01-02T00:00:00.000Z", DateTimeFormatter.ISO_DATE_TIME), PeriodType.SHORT)
         oneOf(printer).close()
       }
     })
 
-    val reply = reportService.printPeriodReport(request.mockRequest(periodReportDTO))
+    val reply = reportService.printPeriodReport(request.mockRequest(getPeriodReport()))
 
     assertThat(reply, isOk)
     assertThat(reply, contains("Report completed"))
   }
 
   @Test
   fun missingCashRegisterForPeriodReport() {
-    val periodReportDTO = ReportService.PeriodReportDTO(LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), "::any ip::", "EXTENDED")
-
     context.checking(object : Expectations() {
       init {
         oneOf(factory).getPrinter("::any ip::")
         will(AbstractExpectations.throwException(DeviceNotFoundException()))
       }
     })
 
-    val reply = reportService.printPeriodReport(request.mockRequest(periodReportDTO))
+    val reply = reportService.printPeriodReport(request.mockRequest(getPeriodReport()))
 
     assertThat(reply, isBadRequest)
     assertThat(reply, contains(ErrorResponse("Device not found.")))
   }
 
   @Test
   fun connectionFailureForPeriodReport() {
-    val periodReportDTO = ReportService.PeriodReportDTO(LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), LocalDateTime.parse("2017-04-23T18:25:43", DateTimeFormatter.ISO_LOCAL_DATE_TIME), "::any ip::", "EXTENDED")
-
     context.checking(object : Expectations() {
       init {
         oneOf(factory).getPrinter("::any ip::")
         will(AbstractExpectations.throwException(IOException()))
       }
     })
 
-    val reply = reportService.printPeriodReport(request.mockRequest(periodReportDTO))
+    val reply = reportService.printPeriodReport(request.mockRequest(getPeriodReport()))
```

Comment:
btw, build() method of the JsonBuilder should build a string value which to be read from transport and to be transformed into an object.